tests/main/interfaces-ssh-keys/task.yaml

```diff
@@ -21,12 +22,18 @@ prepare: |
     echo "testkey" > "$TESTKEY"
     echo "testkey.pub" > "$TESTKEY".pub
 
+    mkdir "$CONFDIR" || true
+    touch "$CONFDIR/test.conf"
+
 restore: |
     rm -rf "$KEYSDIR"
     if [ -d "$KEYSDIR".old ]; then
         mv "$KEYSDIR".old "$KEYSDIR"
     fi
 
+    rm -f "$CONFDIR/test.conf"
+    rmdir "$CONFDIR" || true
```

Comment:
This might remove the system-wide directory if it is empty, which could have side-effects. With KEYSDIR the same could happen but the directory is for a test user that ends up getting added/removed/etc.

I suggest in prepare to use:
```
if [ ! -e "$CONFDIR" ]; do
    mkdir "$CONFDIR"
    touch "$CONFDIR/.spread-created"
fi
```

then in restore:
```
rm -f "$CONFDIR/test.conf"
if [ -e "$CONFDIR/.spread-created" ]; then
    rm -rf "$CONFDIR"
fi
```

Reply:
Updated and squashed. Thanks for the review!

Reply:
> Updated and squashed. Thanks for the review!

Thanks! FYI for future reference, in snapd we typically do not squash during the review process since it makes it more difficult to review past comments. In this case it isn't a big deal since the PR is relatively small.